File: HW1_Task5/src/com/company/Folder.java
package com.company;

import java.util.*;

public class Folder extends AbstractFileSystemNode{

    ArrayList<FileSystemNode> dauthers = new ArrayList<>();
    //конструктор, принимающий список дочерних элементов
    public Folder(FileSystemNode ... fileSystemNodes){
        for(FileSystemNode fileSystemNode : fileSystemNodes){
            dauthers.add(fileSystemNode);
            if(fileSystemNode instanceof AbstractFileSystemNode){
                ((AbstractFileSystemNode) fileSystemNode).parent = this;
            }
        }
    }
    //возвращаем список дочерних элементов
    public ArrayList<FileSystemNode> dauthersOf(){
        return dauthers;
    }
}

File: HW3_Task2/src/Folder.kt
class Folder(vararg fileSystemNodes: FileSystemNode) : AbstractFileSystemNode() {
    private var children = ArrayList<FileSystemNode>()

    init {
        for (fileSystemNode in fileSystemNodes) {
            children.add(fileSystemNode)
            if (fileSystemNode is AbstractFileSystemNode) {
                fileSystemNode.parent = this
            }
        }
    }

    //возвращаем список дочерних элементов
    fun childrenOf(): ArrayList<FileSystemNode> {
        return children
    }
}
